webpack/stories/docs/SlotAndFill.md

```diff
@@ -0,0 +1,116 @@
+# Slot And Fill
+
+`Slot & Fill` allows plugins to extend foreman core functionality in the UI
+
+## Components
+
+### Slot
+`<Slot>` is an opinionated extension point which responsible of rendering its fills
+a Slot can support multiple fills, rendering by weight order by the `multi` prop
+if there is no such prop, it will render the max weighted dfill.
+
+### Fill
+a fill is the filled object of a slot
+
+#### Component fill
+a fill that contains a child  component, which rendered by a dedicated slot
+
+*core*
+```js
+<Slot id='slot-id'>
+ a default child // can be empty
+</Slot>
+```
+
+*plugin A*
+```js
+<Fill id='slot-id' fillId='some-id' weight={100}> 
+  <div> some text </ div>
+</ Fill>
+```
+
+*plugin B*
+```js
+<Fill id='slot-id' fillId='some-id' weight={200}> 
+  <div> some text </ div>
+</ Fill>
+```
+
+Plugin B has a fill with a higher weight, therefore it will be rendered in a dedicated slot.
+
+#### Component fill - remdering multiple fills
+If a slot has a `multi` prop, and it has multiple fills, it will render these fills by weight order
+
+*core*
+```js
+<Slot multi id='slot-id' />
+```
+
+*plugin A*
+```js
+<Fill id='slot-id' fillId='some-id' weight={100}> 
+  <div> some text </ div>
+</ Fill>
+```
+
+*plugin B*
+```js
+<Fill id='slot-id' fillId='some-id' weight={200}> 
+  <div> some text </ div>
+</ Fill>
+```
+
+Plugin B's fill will be render first
+
+#### Props fill
+a fill that contains an `overrideProps` object
+those props are given to the slot's children
+
+
+*core*
+```js
+const TextWrapper = ({ text }) => <div>{text}</div>; 
+
+<Slot id='slot-id'>
+  <TextWrapper text='some default text' />
+</Slot>
+```
+
+*plugin A*
+```js
+<Fill id='slot-id' fillId='some-id' weight={200} overrideProps={{ text: '[Plugin A] this text given by a prop' }} /> 
```

Comment:
Can I use the `overrideProps` in a way it will extend the original prop?
For example, what if I want to add a prefix to the text? so it will end up as `bla bla some default text`?

Reply:
overrideProps by its name will override the slot children props, so basically it replaces the content

Reply:
So how would I implement what I have asked before?
I think this type of example is necessary for the documentation.

Reply:
sure :+1: , Added in the documentation